### ISSUE TITLE
Command sulu:content:resource-locator:maintain not working

### ISSUE THREAD

**Issue by reflexxion** [sulu/sulu#2818]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.3

#### Actual Behavior

When calling `bin/adminconsole sulu:content:resource-locator:maintain` an exception is thrown:
```
 [Symfony\Component\Debug\Exception\FatalThrowableError]
  Type error: Argument 1 passed to Sulu\Component\Content\Types\Rlp\Strategy\RlpStrategy::loadByContent() must implement interface Sulu\Component\Content\Document\B
  ehavior\ResourceSegmentBehavior, instance of Jackalope\Node given, called in vendor/sulu/sulu/src/Sulu/Bundle/ContentBundle/Command/Maintai
  nResourceLocatorCommand.php on line 134
```

#### Steps to Reproduce

Just run `bin/adminconsole sulu:content:resource-locator:maintain`

**Comment by reflexxion** [comment]
Just saw, that it is `@deprecated` but shouldn't they work until removal?

**Comment by danrot** [comment]
The command should definitely work until it has been removed. Apart from that I am not even sure if it should be deprecated...

**Comment by danrot** [comment]
When fixing this we should also think about the `NodeOrderSubscriber`.

**Comment by danrot** [comment]
I would like to also get rid of the `ContentMapper` and `SessionManager` dependencies in the `MaintainResourceLocatorCommand` and the `NodeOrderBuilder`. Therefore it would be a good idea to implement what has already been suggested in https://github.com/sulu/sulu/issues/2772.

**Comment by danrot** [comment]
The `CleanupHistoryCommand` is also not working anymore, because of the publishing changes...

**Comment by danrot** [comment]
Fixed in #2869, #2870, #2873 #2874
